avocado/core/job.py

```diff
@@ -564,6 +570,9 @@ def run(self):
             self.log.error('Report bugs visiting %s', _NEW_ISSUE_LINK)
             return exit_codes.AVOCADO_FAIL
         finally:
+            for ext in self.job_pre_post_dispatcher.extensions:
+                if hasattr(ext.obj, 'post'):
+                    ext.obj.post(self)
```

Comment:
You know too much of Python to understand that it's dynamically typed.  What happens with Python code when pass wrong types? Call "callables" with incorrect signatures? You know all those answer too. 

The point of using Abstract Base Classes, is to *document* interfaces, to the point of telling users what to implement, and how those method implementations signatures should be, but not to enforce it.  Based on previous experience on other (much larger) Python projects[1], this seems to me like a good compromise.

The dispatcher code, on purpose, does not check if a method exists or has the correct signature or even inheritance. [2]  I could be wrong, but I imagine that some people, even yourself, would review code that checked for the correct signatures and say: "hey, this is Python, users are expect to pass the right objects around, if they don't, then crash is what they deserve".

If users want to implement a `JobPre` and use that interface, we'll help by telling him, both with the "code as documentations", with the generated documentation, and the exception on the UI itself, that the `pre` method is needed.  Should we warn on the UI the signature mismatch?

Again, IMHO, this seems like a good compromise.

[1] - http://docs.openstack.org/developer/stevedore/essays/pycon2013.html
[2] - http://docs.openstack.org/developer/stevedore/essays/pycon2013.html#id2